core/services/job/orm.go

```diff
@@ -593,13 +596,37 @@ func (o *orm) FindJobIDsWithBridge(name string) (jids []int32, err error) {
 	return jids, errors.Wrap(err, "FindJobIDsWithBridge failed")
 }
 
-// PipelineRunsByJobsIDs returns pipeline runs for multiple jobs, not preloading data
-func (o *orm) PipelineRunsByJobsIDs(jobsIDs []int32) (runs []pipeline.Run, err error) {
+// FindPipelineRunIDsByJobID fetches the ids of pipeline runs for a job.
+func (o *orm) FindPipelineRunIDsByJobID(jobID int32, offset, limit int) (ids []int64, err error) {
+	err = pg.SqlxTransactionWithDefaultCtx(o.q, o.lggr, func(tx pg.Queryer) error {
```

Comment:
You can use `o.q.Transaction` instead, but I actually think the transaction is not necessary at all here since there is only one query.